bin/varnishd/cache/cache_req_fsm.c

```diff
@@ -695,20 +696,25 @@ cnt_pipe(struct worker *wrk, struct req *req)
 	VCL_pipe_method(req->vcl, wrk, req, bo, NULL);
 
 	switch (wrk->handling) {
-	case VCL_RET_FAIL:
-		req->req_step = R_STP_VCLFAIL;
-		nxt = REQ_FSM_MORE;
-		break;
 	case VCL_RET_SYNTH:
 		req->req_step = R_STP_SYNTH;
 		nxt = REQ_FSM_MORE;
 		break;
 	case VCL_RET_PIPE:
-		AZ(bo->req);
-		bo->req = req;
-		bo->wrk = wrk;
-		SES_Close(req->sp, VDI_Http1Pipe(req, bo));
-		nxt = REQ_FSM_DONE;
+		if (V1P_Enter() == 0) {
+			AZ(bo->req);
+			bo->req = req;
+			bo->wrk = wrk;
+			SES_Close(req->sp, VDI_Http1Pipe(req, bo));
+			nxt = REQ_FSM_DONE;
+			V1P_Leave();
+			break;
+		}
+		wrk->stats->pipe_limited++;
+		/* fall through */
+	case VCL_RET_FAIL:
```

Comment:
why the reordering?

Reply:
To reuse the failure logic by falling through.